src/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -20,7 +21,30 @@ uiRoutes
   .when(DashboardConstants.LANDING_PAGE_PATH, {
     template: dashboardListingTemplate,
     controller: DashboardListingController,
-    controllerAs: 'listingController'
+    controllerAs: 'listingController',
+    resolve: {
+      dash: function ($route, Private, courier, kbnUrl) {
+        const savedObjectsClient = Private(SavedObjectsClientProvider);
+        const title = $route.current.params.title;
+        if (title) {
+          return savedObjectsClient.find({
+            search: `"${title}"`,
+            search_fields: 'title',
+            type: 'dashboard',
+          }).then(results => {
+            // The search isn't an exact match, lets see if we can find a single exact match to use
+            const matchingDashboards = results.savedObjects.filter(dashboard => dashboard.attributes.title === title);
```

Comment:
Do we want this to be case-sensitive? We're prompting the user for duplicate names on save even if they have different cases.

Reply:
my gut feel is that it shouldn't be case sensitive, because I think it'd be a little weird to require them to write in the correct case in order to load the dashboard.    That's just me though, I'd probably guess not case sensitive if I was wondering how it worked.

Reply:
being case-insensitive feels right to me as well